restorer_test.go

```diff
@@ -634,8 +657,79 @@ func testRestorerBuilder(buildpackAPI, platformAPI string) func(t *testing.T, wh
 				})
 
 				it("analyzes with no layer metadata", func() {
-					err := restorer.Restore(testCache)
+					err := restorer.Restore(testCache, nil)
+					h.AssertNil(t, err)
+				})
+			})
+
+			when("there is a previous image", func() {
+				var (
+					fakeAppImage *fakes.Image
+					artifactsDir string
+				)
+
+				it.Before(func() {
+					fakeAppImage = fakes.NewImage(
+						"some-repo/app-image",
+						"some-top-layer-sha",
+						local.IDIdentifier{
+							ImageID: "some-image-id",
+						},
+					)
+
+					var err error
+					artifactsDir, err = ioutil.TempDir("", "lifecycle.artifacts-dir.")
+					h.AssertNil(t, err)
+
+					factory := &layers.Factory{ArtifactsDir: artifactsDir}
+
+					h.AssertNil(t, os.MkdirAll(filepath.Join(layersDir, "sbom", "launch"), os.ModePerm))
+					h.RecursiveCopy(t,
+						filepath.Join("testdata", "restorer", "sbom", "launch"),
+						filepath.Join(layersDir, "sbom", "launch"))
+
+					layer, err := factory.DirLayer("launch.bom", filepath.Join(layersDir, "sbom", "launch"))
 					h.AssertNil(t, err)
+					h.AssertNil(t, fakeAppImage.AddLayerWithDiffID(layer.TarPath, layer.Digest))
+
+					appMetaContents := []byte(`{
+   "buildpacks": [
+       {
+           "key": "buildpack.id",
+           "layers": {
+               "launch-true": {
+                   "data": {},
+                   "launch": true,
+                   "sha": "some-sha"
+               }
+           }
+       }
+   ]
+}
+`)
+
+					h.AssertNil(t, json.Unmarshal(appMetaContents, &restorer.LayersMetadata))
+					restorer.LayersMetadata.BOM = &platform.LayerMetadata{SHA: layer.Digest}
+
+					h.AssertNil(t, os.RemoveAll(filepath.Join(layersDir, "sbom")))
+				})
+
+				it.After(func() {
+					h.AssertNil(t, fakeAppImage.Cleanup())
+					h.AssertNil(t, os.RemoveAll(artifactsDir))
+				})
+
+				it("restores the previous image SBOM layer", func() {
+					h.SkipIf(t, api.MustParse(platformAPI).LessThan("0.8"), "PLATFORM API <0.8 does not restore sBOM")
```

Comment:
```suggestion
					h.SkipIf(t, api.MustParse(platformAPI).LessThan("0.8"), "Platform API < 0.8 does not restore sBOM")
```
```suggestion
					h.SkipIf(t, api.MustParse(platformAPI).LessThan("0.8"), "PLATFORM API <0.8 does not restore sBOM")
```

Reply:
Maybe we should put the `SkipIf` in the before block, that we don't do unnecessary setup?

Reply:
Fixed in latest commit